addons/mail/static/src/components/file_uploader/file_uploader_tests.js

```diff
@@ -87,6 +89,48 @@ QUnit.test('no conflicts between file uploaders', async function (assert) {
     );
 });
 
+QUnit.test('does not crash when an attachment is removed before its upload starts', async function (assert) {
+    assert.expect(1);
+
+    const attachmentUploadedPromise = makeTestPromise();
+    await this.start({
+        async mockFetch(resource, init) {
+            const _super = this._super.bind(this, ...arguments);
+            if (resource === '/web/binary/upload_attachment') {
+                await attachmentUploadedPromise;
+            }
+            return _super();
+        },
+    });
+    const fileUploader = await this.createFileUploaderComponent();
+    const file1 = await createFile({
+        name: 'text1.txt',
+        content: 'hello, world',
+        contentType: 'text/plain',
+    });
+    const file2 = await createFile({
+        name: 'text2.txt',
+        content: 'hello, world',
+        contentType: 'text/plain',
+    });
+    inputFiles(
+        fileUploader.el.querySelector('.o_FileUploader_input'),
+        [file1, file2]
+    );
```

Comment:
Order of attachment being upload look important (while 1st attachment is being uploaded, 2nd attachment is deleted). I think comment in heading of test should be clear enough.